Elicit defaults

markus-kusano opened the pull request:
https://github.com/modelcontextprotocol/go-sdk/issues/617

markus-kusano commented on mcp/mcp_test.go:
```diff
@@ -1459,6 +1461,106 @@ func TestElicitationCapabilityDeclaration(t *testing.T) {
+			expectedValue: "one",
+		},
+		// TODO: should we have a test with number (float) types?
+	}
```
Invalid schemas  are tested on line 1137 above, were there some other testcases you had in mind?

findleyr replied:
Ah, that was largely beneath the fold.

Maybe add a case or two more related to the new default values? How about an invalid number and enum?

markus-kusano commented:
PTAL, I updated the server side to both apply defaults and do validation. I also removed the use of reflect as discussed offline (thanks for the help!)

findleyr commented on mcp/client.go:
```diff
@@ -341,6 +344,9 @@ func validateElicitSchema(wireSchema any) (*jsonschema.Schema, error) {
 	if err := remarshal(wireSchema, &schema); err != nil {
 		return nil, err
 	}
+	if schema == nil {
```
On the server the schema should be a *jsonschema.Schema, not a **jsonschema.Schema. Where are you seeing this?

Let's merge this as-is, and I will take a look.

liut commented on mcp/server.go:
```diff
@@ -1015,7 +1015,37 @@ func (ss *ServerSession) Elicit(ctx context.Context, params *ElicitParams) (*Eli
+
+	resolved, err := schema.Resolve(nil)
+	if err != nil {
+		fmt.Printf("  resolve err: %s", err)
```
Is this okay?
